pkg/migrations/sql/idempotent/base.sql

```diff
@@ -1931,6 +1950,72 @@ SET search_path = pg_temp;
 REVOKE ALL ON FUNCTION SCHEMA_CATALOG.create_metric_view(text) FROM PUBLIC;
 GRANT EXECUTE ON FUNCTION SCHEMA_CATALOG.create_metric_view(text) TO prom_writer;
 
+CREATE OR REPLACE FUNCTION SCHEMA_CATALOG.register_metric_view(schema_name name, view_name name)
+    RETURNS BOOLEAN
+AS $func$
+DECLARE
+   metric_table_name name;
+BEGIN
+    -- check if table/view exists
+    PERFORM * FROM information_schema.tables 
+    WHERE  table_schema = register_metric_view.schema_name
+    AND    table_name   = register_metric_view.view_name;
+
+    IF NOT FOUND THEN
+        RAISE EXCEPTION 'cannot register non-existant metric view in specified schema';
+    END IF;
+
+    -- cannot register view in data schema
+    IF schema_name = 'SCHEMA_DATA' THEN
+        RAISE EXCEPTION 'cannot register metric view in data schema';
+    END IF; 
+
+    -- check if view is based on a metric from prom_data
+    SELECT table_name FROM information_schema.view_table_usage
+    INTO metric_table_name
+    WHERE view_schema = register_metric_view.schema_name
+    AND view_table_usage.view_name = register_metric_view.view_name
+    AND table_schema = 'SCHEMA_DATA';
+
+    IF NOT FOUND THEN
+        RAISE EXCEPTION 'view not based on a metric table from SCHEMA_DATA schema';
+    END IF;
+
+    -- check if the view columns match columns from the underlying metric table
+    PERFORM m.column_name 
+    FROM 
+        (SELECT column_name, data_type FROM information_schema.columns 
+        WHERE  table_schema = 'SCHEMA_DATA'
+        AND    table_name   = metric_table_name
+        EXCEPT
+        SELECT column_name, data_type FROM information_schema.columns 
+        WHERE  table_schema = register_metric_view.schema_name
+        AND    table_name   = register_metric_view.view_name) AS m;
+
+    IF FOUND THEN
+        RAISE EXCEPTION 'view columns must match names and data types with the underlying metric table';
+    END IF;
+
+    -- insert into metric table
+    INSERT INTO SCHEMA_CATALOG.metric (metric_name, table_name, table_schema, series_table, is_view, creation_completed)
+    VALUES (register_metric_view.view_name, register_metric_view.view_name, register_metric_view.schema_name, metric_table_name, true, true)
+    ON CONFLICT DO NOTHING;
+
+    IF NOT FOUND THEN
+        RAISE EXCEPTION 'metric with the same name and schema already exists, could not register';
```

Comment:
we probably want an optional argument to this function  "if_not_exists => false" and we don't throw this error if it's true. This would allow people to use this function in a idempotent way